core/arch/arm/kernel/boot.c

```diff
@@ -1512,3 +1578,41 @@ void __weak boot_save_args(unsigned long a0, unsigned long a1 __unused,
 		}
 	}
 }
+
+#if defined(CFG_TRANSFER_LIST)
+static TEE_Result release_transfer_list(void)
+{
+	struct dt_descriptor *dt = get_external_dt_desc();
+
+	if (!mapped_tl)
+		return TEE_SUCCESS;
+
+	if (dt) {
+		int ret = 0;
+		struct transfer_list_entry *te = NULL;
+
+		/*
+		 * Pack the DTB and update the transfer list before un-mapping
+		 */
+		ret = fdt_pack(dt->blob);
+		if (ret < 0) {
+			EMSG("Failed to pack Device Tree at 0x%" PRIxPA
+			     ": error %d", virt_to_phys(dt->blob), ret);
+			panic();
+		}
+
+		te = transfer_list_find(mapped_tl, TL_TAG_FDT);
+		assert(dt->blob == transfer_list_entry_data(te));
+		transfer_list_set_data_size(mapped_tl, te,
+					    fdt_totalsize(dt->blob));
+		dt->blob = NULL;
+	}
+
+	transfer_list_unmap_sync(mapped_tl);
+	mapped_tl = NULL;
+
+	return TEE_SUCCESS;
+}
+
+boot_final(release_transfer_list);
```

Comment:
`release_external_dt()` from _core/kernel/dt.c_ also packs and unmaps the external DTB and is also registered as a `boot_final` initcall handler. There may be conflict between the 2 depending on which is called first.

Reply:
There is no issue about that, please check the latest commit, `release_external_dt` only do packing and un-mapping when the DTB memory is mapped in `MEM_AREA_EXT_DT`, when it find out that is mapped in other areas, it skips both packing and un-mapping so there will be no conflicts when `CFG_TRANSFER_LIST` is enabled.